uniro_admin_frontend/src/component/LogMap.tsx

```diff
@@ -0,0 +1,328 @@
+        for (const coreRoutes of coreRouteList) {
+            const { coreNode1Id, coreNode2Id, routes: subRoutes } = coreRoutes;
+
+            const routeIds = subRoutes.map((el) => el.routeId);
+
+            const key = coreNode1Id < coreNode2Id ? routeIds.join("_") : routeIds.reverse().join("_");
+
+            if (cachedRoute.current.has(key)) {
+                const cachedPolyline = cachedRoute.current.get(key);
+                if (!cachedPolyline) continue;
+
+                cachedPolyline.setMap(map);
+                cachedPolyline.setOptions({
+                    strokeColor: "#808080",
+                });
+                continue;
+            }
+
+            const subNodes = [subRoutes[0].node1, ...subRoutes.map((el) => el.node2)];
+
+            const routePolyLine = new Polyline({
+                map: map,
+                path: subNodes.map((el) => {
+                    return { lat: el.lat, lng: el.lng };
+                }),
+                strokeColor: "#808080",
+                geodesic: true,
+            });
+
+            cachedRoute.current.set(key, routePolyLine);
+        }
+    };
+
+    const findCachedCoreRoute = (coreRouteList: CoreRoutes[]) => {
+        if (!Polyline) return;
+
+        for (const coreRoutes of coreRouteList) {
+            const { coreNode1Id, coreNode2Id, routes } = coreRoutes;
+
+            const routeIds = routes.map((el) => el.routeId);
+
+            const key = coreNode1Id < coreNode2Id ? routeIds.join("_") : routeIds.reverse().join("_");
+
+            if (cachedRoute.current.has(key)) {
+                const polyline = cachedRoute.current.get(key);
+
+                if (!polyline) return;
+
+                polyline.setMap(map);
+                polyline.setOptions({
+                    strokeColor: "red",
+                });
+
+                continue;
+            }
+
+            const subNodes = [routes[0].node1, ...routes.map((el) => el.node2)];
+
+            const routePolyLine = new Polyline({
+                map: map,
+                path: subNodes.map((el) => {
+                    return { lat: el.lat, lng: el.lng };
+                }),
+                strokeColor: "red",
+                geodesic: true,
+            });
+
+            cachedRoute.current.set(key, routePolyLine);
+        }
+    };
+
+    const typeofChanged = (changed: ChangedRouteType): RevisionMarkerTypes => {
+        const { current: 최신버전, difference: 현재버전 } = changed;
+
+        const 최신버전_요소_개수 = 최신버전.dangerFactors.length + 최신버전.cautionFactors.length;
+        const 현재버전_요소_개수 = 현재버전.dangerFactors.length + 현재버전.cautionFactors.length;
+
+        if (최신버전_요소_개수 === 0) return "REMOVED";
+
+        if (현재버전_요소_개수 === 0) return "CREATED";
+
+        return "CHANGED";
+    };
+
+    const addChangedMarkers = (changedList: (ChangedRouteWithNodeType | undefined)[]) => {
+        if (!AdvancedMarker) return;
```

Comment:
map에 대한 방어코드를 넣는 건 어떨까요?? 하위에 Not-Null Assertion을 활용하는 부분이 많이 존재하는 것으로 보이는데, 해당 검증으로 더욱 안전한 코드가 될 것 같습니다
```suggestion
        if (!AdvancedMarker || !map) return;
```